pkg/gateway/operations/postobject.go

```diff
@@ -110,7 +111,8 @@ func (controller *PostObject) HandleCompleteMultipartUpload(w http.ResponseWrite
 	}
 
 	scheme := httputil.RequestScheme(req)
-	location := fmt.Sprintf("%s://%s/%s/%s/%s", scheme, o.FQDN, o.Repository.Name, o.Reference, o.Path)
+	location := fmt.Sprintf("%s://%s/api/v1/repositories/%s/refs/%s/objects?path=%s",
+		scheme, req.Host, url.PathEscape(o.Repository.Name), url.PathEscape(o.Reference), url.QueryEscape(o.Path))
```

Comment:
Isn't this a breaking change?

Reply:
true, adding a bigger comment at the end of the pr why this alt. solution is not good.